app/validators/email_domain_validator.rb

```diff
@@ -0,0 +1,21 @@
+class EmailDomainValidator < ActiveModel::Validator
+  ALLOWED_DOMAINS = [
+    'justice.gov.uk',
+    'digital.justice.gov.uk'
+  ].freeze
+
+  def validate(record)
+    return if record.email.blank?
+
+    unless allowed_domain?(record.email)
+      record.errors.add(:base, I18n.t('activemodel.errors.models.from_address.invalid_domain'))
+    end
+  end
+
+  private
+
+  def allowed_domain?(email)
+    domain = email.split('@').last
```

Comment:
```suggestion
    domain = Mail::Address.new(email).domain
```

Reply:
ooo this is handy! Had no idea this existed.

Reply:
Unfortunately the `Mail::Address` class requires a correctly formatted email, otherwise it throws an error. One way around this is to maybe add a proc to run if the email has passed the email format validation?

```
 validates_with EmailDomainValidator, if: proc { |obj| obj.errors.empty? }, allow_blank: true
```